pkg/machine/qemu/machine.go

```diff
@@ -181,6 +206,30 @@ func (v *MachineVM) Start(name string, _ machine.StartOptions) error {
 	}
 
 	_, err = os.StartProcess(v.CmdLine[0], cmd, attr)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Waiting for VM ...")
+	socketPath, err := getSocketDir()
+	if err != nil {
+		return err
+	}
+
+	// The socket is not made until the qemu process is running so here
+	// we do a backoff waiting for it.  Once we have a conn, we break and
+	// then wait to read it.
+	for i := 0; i < 6; i++ {
+		conn, err = net.Dial("unix", filepath.Join(socketPath, "podman", v.Name+"_ready.sock"))
+		if err == nil {
+			break
+		}
+		time.Sleep(wait)
+		wait++
+	}
```

Comment:
We might also need to add this to the stop logic - if a `podman machine stop` is run before the machine is started and in a proper state, we will get
```
Failed to get "write" lock
Is another process using the image [/home/acui/.local/share/containers/podman/machine/qemu/ah_fedora-coreos-33.20210315.1.0-qemu.x86_64.qcow2]?
```

And this can only be solved via removing the vm or killing the qemu process.

This can be done in another PR if you'd like though

Reply:
nice spot @ashley-cui, take the next two days off on me!  :smiley: